File: atcoder/abc147B_palindrome_philia.rs
// Vicfred
// https://atcoder.jp/contests/abc147/tasks/abc147_b
// string manipulation
use std::io;

fn main() {
    let mut s = String::new();

    io::stdin()
        .read_line(&mut s)
        .unwrap();

    let s = s.trim();

    let n = s.len();
    let s: Vec<char> = s.chars().collect();

    let mut answer = 0;

    for i in 0..n as usize {
        if s[i] != s[n-i-1] {
            answer += 1;
        }
    }

    println!("{}", answer/2);

}

File: atcoder/abc147B_palindrome_philia_.ml
(*(* vicfred
 * https://atcoder.jp/contests/abc147/tasks/abc147_b
 * string manipulation
 * *)
let s, ans = read_line(), ref 0
let main =
    String.(iteri (fun i c -> if s.[length s - i - 1] <> c then incr ans) s);
    Printf.printf "%d\n" @@ !ans / 2

